mobile/mobile-core/android-device-discovery.ts

```diff
@@ -0,0 +1,60 @@
+///<reference path="./../../.d.ts"/>
+"use strict";
+
+import {DeviceDiscovery} from "./device-discovery"
+import * as helpers from "../../helpers";
+import {AndroidDevice} from "../android/android-device";
+import {EOL} from "os";
+
+export class AndroidDeviceDiscovery extends DeviceDiscovery implements Mobile.IAndroidDeviceDiscovery {
+	private _devices: string[] = [];
+
+	constructor(private $childProcess: IChildProcess,
+		private $injector: IInjector,
+		private $staticConfig: Config.IStaticConfig) {
+		super();
+	}
+
+	private createAndAddDevice(deviceIdentifier: string): void {
+		this._devices.push(deviceIdentifier);
+		let device = this.$injector.resolve(AndroidDevice, { identifier: deviceIdentifier });
+		this.addDevice(device);
+	}
+
+	private deleteAndRemoveDevice(deviceIdentifier: string): void {
+		_.remove(this._devices, d => d === deviceIdentifier);
+		this.removeDevice(deviceIdentifier);
+	}
+
+	public startLookingForDevices(): IFuture<void> {
+		return(()=> {
+			let requestAllDevicesCommand = `"${this.$staticConfig.getAdbFilePath().wait()}" devices`;
```

Comment:
ok, this is not a merge stopper, I just don't like so long call in double quotes on two places in the same class:
```TypeScript
"${this.$staticConfig.getAdbFilePath().wait()}"
```